github/Branch.py

```diff
@@ -327,14 +339,29 @@ def edit_required_status_checks(
         self,
         strict: Opt[bool] = NotSet,
         contexts: Opt[list[str]] = NotSet,
+        checks: List[Union[str, Tuple[str, int]]] = NotSet,
     ) -> RequiredStatusChecks:
         """
         :calls: `PATCH /repos/{owner}/{repo}/branches/{branch}/protection/required_status_checks <https://docs.github.com/en/rest/reference/repos#branches>`_
         """
         assert is_optional(strict, bool), strict
         assert is_optional_list(contexts, str), contexts
+        # assert is_optional_list(checks, Union[str, Tuple[str, int]]), checks
+
+        post_parameters: dict[str, Any] = NotSet.remove_unset_items({"strict": strict})
+
+        if is_defined(checks):
+            post_parameters["checks"] = []
+            for check in checks:
+                if isinstance(check, tuple):
+                    context, app_id = check
+                    post_parameters["checks"].append({"context": context, "app_id": app_id})
+                else:
+                    post_parameters["checks"].append({"context": check})
+
+        elif is_defined(contexts):
+            post_parameters["checks"] = [{"context": context} for context in contexts]
 
-        post_parameters: dict[str, Any] = NotSet.remove_unset_items({"strict": strict, "contexts": contexts})
```

Comment:
please keep the `NotSet.remove_unset_items` pattern where it exists, preferred over `is_defined + post_parameters = ... above`.

Reply:
tried to comply with the pattern but didn't found a example in PyGithub which uses kind of calculation of the parameters like I do here.
I dont think https://github.com/PyGithub/PyGithub/pull/2884/commits/8776ae5e421d5b7a739d9113e63ab05f97f01efd makes it easier to read, maybe you could make it clearer what I have to adjust based on that.